### PR TITLE
Issue#81, added "play" function in "Chapter" mode

### PR DESCRIPTION
**Pull request by stormdb**
Added "play" function in "Chapter" mode

### CODE REVIEW

File: src/main/java/uk/yermak/audiobookconverter/fx/FilesController.java [hunk 1]
```diff
@@ -150,6 +150,17 @@ public void initialize() {
         filesChapters.getTabs().remove(chaptersTab);
 
         bookStructure.getSelectionModel().setSelectionMode(SelectionMode.MULTIPLE);
+        bookStructure.getSelectionModel().getSelectedItems().addListener((ListChangeListener<TreeItem<Organisable>>) c -> {
+            List<MediaInfo> list = ConverterApplication.getContext().getSelectedMedia();
+            list.clear();
+            for (TreeItem<Organisable> item : c.getList()) {
+                if (item.getValue() instanceof Chapter chapter) {
+                    list.addAll(chapter.getMedia());
+                } else if (item.getValue() instanceof MediaInfoProxy mediaInfoProxy) {
```
Comment (yermak) [inline src/main/java/uk/yermak/audiobookconverter/fx/FilesController.java:159]: MediaInfo Interface should be used rather then Proxy which should never been accessed directly.

File: src/main/java/uk/yermak/audiobookconverter/fx/FilesController.java [hunk 2]
```diff
@@ -150,6 +150,17 @@ public void initialize() {
         filesChapters.getTabs().remove(chaptersTab);
 
         bookStructure.getSelectionModel().setSelectionMode(SelectionMode.MULTIPLE);
+        bookStructure.getSelectionModel().getSelectedItems().addListener((ListChangeListener<TreeItem<Organisable>>) c -> {
+            List<MediaInfo> list = ConverterApplication.getContext().getSelectedMedia();
+            list.clear();
+            for (TreeItem<Organisable> item : c.getList()) {
```
Comment (yermak) [inline src/main/java/uk/yermak/audiobookconverter/fx/FilesController.java:156]: Is it better solution to extend interface Organisable with method getMedia and avoid instance check completely?

File: src/main/java/uk/yermak/audiobookconverter/fx/FilesController.java [hunk 3]
```diff
@@ -150,6 +150,17 @@ public void initialize() {
         filesChapters.getTabs().remove(chaptersTab);
 
         bookStructure.getSelectionModel().setSelectionMode(SelectionMode.MULTIPLE);
+        bookStructure.getSelectionModel().getSelectedItems().addListener((ListChangeListener<TreeItem<Organisable>>) c -> {
+            List<MediaInfo> list = ConverterApplication.getContext().getSelectedMedia();
```
Comment (yermak) [inline src/main/java/uk/yermak/audiobookconverter/fx/FilesController.java:154]: Question more to myself... If selected media used only for playback is there any better place to keep them or should this var have better name?
I personally though that this feature will require media player modification, but that's interesting if it works this way...

**Review by yermak — CHANGES_REQUESTED** [review]
Added few comments for consideration, only Proxy related is critical other for your consideration...

File: src/main/java/uk/yermak/audiobookconverter/fx/FilesController.java (hunk 2 above)
Comment (stormdb) [reply src/main/java/uk/yermak/audiobookconverter/fx/FilesController.java]: Replaced with Organizable.getMedia()

File: src/main/java/uk/yermak/audiobookconverter/fx/FilesController.java (hunk 1 above)
Comment (stormdb) [reply src/main/java/uk/yermak/audiobookconverter/fx/FilesController.java]: Fixed

File: src/main/java/uk/yermak/audiobookconverter/fx/FilesController.java (hunk 3 above)
Comment (stormdb) [reply src/main/java/uk/yermak/audiobookconverter/fx/FilesController.java:154]: Name seems to reflect the purpose - list of selected media items.

File: src/main/java/uk/yermak/audiobookconverter/MediaInfoProxy.java [hunk 4]
```diff
@@ -61,6 +63,11 @@ public void moveDown() {
         this.getMediaInfo().moveDown();
     }
 
+    @Override
+    public List<MediaInfo> getMedia() {
+        return List.of(getMediaInfo());
```
Comment (yermak) [inline src/main/java/uk/yermak/audiobookconverter/MediaInfoProxy.java:68]: Could you please change to delegate please, as proxy should not bring additional logic just wrapping

**Review by yermak — CHANGES_REQUESTED** [review]
Please change Proxy class

File: src/main/java/uk/yermak/audiobookconverter/MediaInfoProxy.java (hunk 4 above)
Comment (stormdb) [reply src/main/java/uk/yermak/audiobookconverter/MediaInfoProxy.java]: Fixed

**Review by yermak — APPROVED** [review]
Thank you

**Comment by stormdb** [comment]
Closes #81

**Pull request merged**